search-provider/eks-discovery-feed-provider.c

```diff
@@ -435,6 +469,166 @@ handle_article_card_descriptions (EksDiscoveryFeedDatabaseContentProvider *skele
 }
 
 static void
+get_word_of_the_day_content_cb (GObject *source,
+                                GAsyncResult *result,
+                                gpointer user_data)
+{
+  EkncEngine *engine = EKNC_ENGINE (source);
+  DiscoveryFeedQueryState *state = user_data;
+
+  g_application_release (g_application_get_default ());
+
+  GError *error = NULL;
+  g_autoptr(GSList) models = NULL;
+  GSList *shards = NULL;
+
+  if (!models_and_shards_for_result (engine,
+                                     state->provider->application_id,
+                                     result,
+                                     &models,
+                                     &shards,
+                                     &error))
+    {
+      g_dbus_method_invocation_take_error (state->invocation, error);
+      discovery_feed_query_state_free (state);
+      return;
+    }
+
+  GVariantBuilder builder;
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{ss}"));
+
+  gint index = get_day_of_year () % g_slist_length (models);
+  EkncContentObjectModel *model = g_slist_nth (models, index)->data;
```

Comment:
Possibly not a big deal, but `g_slist_length` and `g_slist_nth` are both O(N). It might be good to look into exposing an `eknc_query_object_get_n_models` member function which returns the list length directly without having to measure it, so that we can calculate the index as we do above, then use `g_slist_nth`